File: insertion_sort.c
#include<stdio.h>
int i;
int j;
void selectionsort(int n , int arr[]){
   
    for(i=1;i<n;i++){
        int key = arr[i];
        int j=i-1;
      while(j>=0 && arr[j]>key){
          
         arr[j+1] = arr[j];
         j--;
      }
      arr[j+1] = key;
    }
}
void printarr(int arr[],int n){
    for(i=0;i<n;i++){
        printf("%d",arr[i]);
    }
    printf("\n");
}
int main()
{
    int n;
    int i,j;
    printf("Enter the number of elements in an array : ");
    scanf("%d", &n);
    int arr[n];
    for(i=0;i<n;i++){
    printf("Enter the %dth element in an array : ", i+1);
    scanf("%d", &arr[i]);
    }
    printf("The Unsorted Array is : ");
    printf("\n");
    printarr( arr, n);

    selectionsort(n,arr);
    printf("The Sorted Array is : ");
    printf("\n");
     printarr( arr, n);
     return 0;
}

File: Insertion_sort.java
import java.util.Scanner;

public class Insertion_sort {
 void Insertion (int[] arr)
{
    int n = arr.length;
    int i;

   
    for(i=1;i<n;i++){
        int key = arr[i];
        int j=i-1;
        while (j>=0 && arr[j] > key)
         {
            arr[j+1] = arr[j];
            j=j-1;
            
            

        }
        arr[j+1] = key;   
        
        
    }
}

public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    System.out.print("Enter the number of terms in an array = ");
    
    int n = sc.nextInt();
    int i;
    int [] arr = new int[n];
    for(i=0;i<n;i++){
        System.out.print("Enter the " +(i+1)+ " th in an array = ");
        arr[i] = sc.nextInt();
    }
    Insertion_sort sorter = new Insertion_sort();
     sorter.Insertion(arr);

     System.out.print("The Sorted array is: ");
        for (int o = 0; o < n;o++) {
            System.out.print(arr[o] + " ");

    
}
}
}
